### PR TITLE
MUZIMA-330 - Form should require descriminator tags

### PR DESCRIPTION
**Pull request by yoctopus**
MUZIMA 330

To enable that functionality. I have added a method to the AvailableForm class to check the discriminator against another discriminator passed for check up.

In the AvailableFormBuilder class, I have set a discriminator from the discriminator in the base Form class.
In the Form TemplateWizardActivity, I have utilised the method in first above to check if this AvailableForm is a registration form using a discriminator for Registration forms passed into the method.

### CODE REVIEW

**Comment by yoctopus** [comment]
Please reach me through 'vinsonpeter8@gmail.com' for more clarification.
Regards.
Thank you.

File: app/src/main/java/com/muzima/view/forms/FormTemplateWizardActivity.java [hunk 1]
```diff
@@ -149,7 +150,7 @@ private boolean hasRegistrationFormSelected() {
         for (int i = 0; i < checkedItemPositions.size(); i++) {
             if (checkedItemPositions.valueAt(i)) {
                 AvailableForm selectedForm = (AvailableForm) listView.getItemAtPosition(checkedItemPositions.keyAt(i));
-                if (selectedForm.isRegistrationForm()) {
+                if (selectedForm.hasDescriminator(Constants.FORM_DISCRIMINATOR_REGISTRATION)) {
```
Comment (mssavai) [inline app/src/main/java/com/muzima/view/forms/FormTemplateWizardActivity.java]: Could you redefine the logic for determining whether it's a registration form within the isRegistrationForm() method? The reason is, that method can used in several places and therefore we need to remain consistent.

File: app/src/main/java/com/muzima/model/AvailableForm.java [hunk 2]
```diff
@@ -42,4 +43,17 @@ public boolean isRegistrationForm(){
         }
         return false;
     }
+    public boolean hasDescriminator(String discriminator){
```
Comment (mssavai) [inline app/src/main/java/com/muzima/model/AvailableForm.java]: I think you could simply compare supplied discriminator against getDiscriminator().

**Comment by yoctopus** [comment]
Hi. I have updated, generated the new apk and attached to the ticket.
You may check out.

**Pull request closed without merging**